dissect/target/plugins/os/windows/tasks.py

```diff
@@ -58,452 +61,124 @@
+                    self.task_objects.append(XmlTask(entry, target))
 
     def check_compatible(self):
-        if len(self.files) == 0:
+        if len(self.task_objects) == 0:
             raise UnsupportedPluginError("No task files")
 
-    @export(record=DynamicDescriptor(["uri", "datetime"]))
-    def tasks(self):
+    @export(record=DynamicDescriptor(["path", "datetime"]))
+    def tasks(self) -> Iterator:
         """Return all scheduled tasks on a Windows system.
 
         On a Windows system, a scheduled task is a program or script that is executed on a specific time or at specific
         intervals. An adversary may leverage such scheduled tasks to gain persistence on a system.
 
         References:
-            - https://en.wikipedia.org/wiki/Windows_Task_Scheduler
-        """
-        for f in self.files:
-            try:
-                for entry in self.parse_task(f):
-                    yield entry
-            except Exception as e:
-                self.target.log.warning("An error occured parsing task %s: %s", f, str(e))
-                self.target.log.debug("", exc_info=e)
-
-    def parse_task(self, entry):
-        try:
-            task_xml = strip_namespace(ElementTree.fromstring(entry.open().read(), forbid_dtd=True))
-        except ElementTree.ParseError:
-            raise InvalidTaskError()
-
-        # RegistrationInfo
-        task = Task(task_xml)
-        task.hostname = self.target.hostname
-        task.uri = task.get_element("RegistrationInfo/URI")
-        task.security_descriptor = task.get_element("RegistrationInfo/SecurityDescriptor")
-        task.source = task.get_element("RegistrationInfo/Source")
-        task.date = task.get_element("RegistrationInfo/Date")
-        task.author = task.get_element("RegistrationInfo/Author")
-        task.version = task.get_element("RegistrationInfo/Version")
-        task.description = task.get_element("RegistrationInfo/Description")
-        task.documentation = task.get_element("RegistrationInfo/Documentation")
+            https://en.wikipedia.org/wiki/Windows_Task_Scheduler
 
-        # Principals
-        task.principal_id = task_xml.find("Principals/Principal").get("id")
-        task.user_id = task.get_element("Principals/Principal/UserId")
-        task.logon_type = task.get_element("Principals/Principal/LogonType")
-        task.group_id = task.get_element("Principals/Principal/GroupId")
-        task.display_name = task.get_element("Principals/Principal/DisplayName")
-        task.run_level = task.get_element("Principals/Principal/RunLevel")
-        task.process_token_sid_type = task.get_element("Principals/Principal/ProcessTokenSidType")
-        task.required_privileges = task.get_element("Principals/Principal/RequiredPrivileges")
+        Yields:
+            The scheduled tasks found on the target.
+        """
+        for task_object in self.task_objects:
+            yield from self.get_task_fields(task_object)
```

Comment:
This function doesn't really do anything now. Make `_get_task_fields` just return the fields dict from a `Task*` object and put the reset of the logic in here.